interceptors/logging/interceptors.go

```diff
@@ -45,42 +48,94 @@ func (c *reporter) PostCall(err error, duration time.Duration) {
 	code := c.opts.codeFunc(err)
 	logger = logger.With("grpc.code", code.String())
 	if err != nil {
-		logger = logger.With("error", fmt.Sprintf("%v", err))
+		logger = logger.With("grpc.error", fmt.Sprintf("%v", err))
+	}
+
+	logger = logger.With(c.opts.durationFieldFunc(duration)...)
+	logger = logger.With("protocol", "grpc")
+	logger.Log(c.opts.levelFunc(code), fmt.Sprintf("finished_%s_%s_call", c.kind, c.typ))
+}
+
+// PostMsgSend logs the details of the servingObject that is flowing out of the rpc.
+// resp object wrt server.
+// Log the details of the first request, skip if the object is response.
+func (c *reporter) PostMsgSend(resp interface{}, err error, duration time.Duration) {
+	if !c.opts.shouldLog(interceptors.FullMethod(c.service, c.method), err) {
```

Comment:
I think we might want bit more control, have some idea. (:

Reply:
Elaborate please

Reply:
What are the options you wany for the decider?

Reply:
We agreed on the decider